pkg/clustermanager/provider.go

```diff
@@ -136,10 +148,13 @@ func (p *provider) processProviderWatchEvents() {
 				lcluster.Spec.Managed = false
 				lcluster.Status.ClusterConfig = event.LCInfo.Config
 				lcluster.Status.Phase = lcv1alpha1apis.LogicalClusterPhaseReady
+				lcluster.ObjectMeta.Finalizers = append(reflcluster.ObjectMeta.Finalizers, finalizerName)
```

Comment:
why are we adding finalizers to unmanaged ?

Reply:
We shouldn't.